content/legal/generative-tooling.md

```diff
@@ -19,32 +19,38 @@ As explained by the following [U.S. Copyright Office Registration Guidance (3/16
 
-These portions authored by a human may simply come from the prompt the human provided or subsequent changes they make. However, a prominent concern with generative AI is the risk of reproducing portions of materials that they were trained on, some of which may be copyrightable subject matter. Thus, a recommended practice when using generative AI tooling is to use tools with features that identify any included content that is similar to parts of the tool’s training data, as well as the license of that content.
+These portions authored by a human may simply come from the prompt the human provided or subsequent changes they make. However, a prominent concern with generative AI is the risk of reproducing copyrightable portions of materials that they were trained on, some of which may be copyrightable subject matter. Thus, a recommended practice when using generative AI tooling is to use tools with features that identify any included content that is similar to parts of the tool’s training data, as well as the license of that content.
 
 Given the above, code generated in whole or in part using AI can be contributed if the contributor ensures that:
 
-1. The terms and conditions of the generative AI tool do not place any restrictions on use of the output that would be inconsistent with the Open Source Definition (e.g., ChatGPT’s terms are inconsistent).
+1. The terms and conditions of the generative AI tool do not place any restrictions on use of the output that would be inconsistent with the [Open Source Definition](https://opensource.org/osd/).
 2. At least one of the following conditions is met:
-    1. The output is not copyrightable subject matter (and would not be even if produced by a human)
-    2. No third party materials are included in the output
-    3. Any third party materials that are included in the output are being used with permission (e.g., under a compatible open source license) of the third party copyright holders and in compliance with the applicable license terms
-3. A contributor obtain reasonable certainty that conditions 2.2 or 2.3 are met if the AI tool itself provides sufficient information about materials that may have been copied, or from code scanning results
-    1. E.g. AWS CodeWhisperer recently added a feature that provides notice and attribution
+    1. The output is not copyrightable subject matter (and would not be even if produced by a human).
+    2. No third party materials are included in the output.
+    3. Any third party materials that are included in the output are being used with permission (e.g., under a compatible open-source license) of the third party copyright holders and in compliance with the applicable license terms.
+3. A contributor obtain reasonable certainty that conditions 2.2 or 2.3 are met if the AI tool itself provides sufficient information about output that may be similar to training data, or from code scanning results.
```

Comment:
On item 3, ...What are these "code scanning" results? 

Does that mean someone reads the code and expresses an assessment, or is this a reference to some class of code scanner tools that I don't know exist, but that supposedly provide stronger assurance about whether and which copyrighted works are being infringed by what lines of code?

"code scanning results" suggests tools are needed, and if that is the intent you need to be more specific about the class of such tools that can do this job. 

But for quite a lot of gen-AI output, I think I can have "reasonable certainty" just by reading the code.

Reply:
I did not change the 'code scanning' part of the current guidance so I won't go deep on the reasoning behind that part of the text. Maybe @rvs can give you a better context.